### PR TITLE
Mjolnier's throw gets countered by anti-magic

### PR DESCRIPTION
**Pull request by Chubbygummibear**
# Document the changes in your pull request

And by "countered", i mean it ~~explodes~~ turns into a statue on the floor that then has to be picked up again by the wizard or crew

I've definitely encountered the wizard Mjolnier + lich + recall combo or whatever it is that gives the wizard the ability to respawn and keep pulling mjolnier back from wherever they threw it to, and public chat was malding over it yesterday as well so that feels like enough justification to use my time making this.

Whacking people with it while two-handing it is unchanged, you still smack and get your free egregiously long stun window as a reward for putting yourself in some danger by getting close to others.

But if you try to play it safe by spam throwing it, and it connects with someone who has anti magic...
image
image
image
image

Mjolnir becomes a passable statue that needs to be interacted with to get it back. takes 30 seconds of standing still for crew to get it back, and 15 seconds for wizards or magically aligned beings

# Wiki Documentation

~~Chaplain mains eating~~
Mjolnir throw is countered by anti-magic

# Changelog

Edit the changelog below to reflect the changes made by this PR, even if the changes are minor - required for every PR.
If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name.

:cl:  
rscadd: Throwing Mjolnier into someone with anti-magic causes the hammer to go inert and fall to the ground as a monument that can then be picked up again by someone who pulls hard enough to unearth it
/:cl:

### CODE REVIEW

File: code/game/objects/items/singularityhammer.dm [hunk 1]
```diff
@@ -85,6 +85,9 @@
 	throwforce = 30
 	throw_range = 7
 	w_class = WEIGHT_CLASS_HUGE
+	var/tesla_power = 20000
```
Comment (ToasterBiome) [inline code/game/objects/items/singularityhammer.dm]:
```suggestion
	var/zap_power= 20000
```

yes while technically you are using tesla_zap, it isn't from a tesla so it shouldn't reference one.

File: code/game/objects/items/singularityhammer.dm [hunk 2]
```diff
@@ -85,6 +85,9 @@
...
+	var/tesla_range = 20
```
Comment (ToasterBiome) [inline code/game/objects/items/singularityhammer.dm]:
```suggestion
	var/zap_range = 20
```

File: code/game/objects/items/singularityhammer.dm [hunk 3]
```diff
@@ -85,6 +85,9 @@
...
+	var/tesla_flags = TESLA_MOB_DAMAGE | TESLA_MOB_STUN | TESLA_OBJ_DAMAGE
```
Comment (ToasterBiome) [inline code/game/objects/items/singularityhammer.dm]:
```suggestion
	var/zap_flags = TESLA_MOB_DAMAGE | TESLA_MOB_STUN | TESLA_OBJ_DAMAGE
```

File: code/game/objects/items/singularityhammer.dm [hunk 4]
```diff
@@ -107,7 +110,20 @@
...
-		shock(hit_atom)
+		var/mob/M = hit_atom
+		var/atom/A = M.anti_magic_check()
+		if(A)
+			if(isitem(A))
+				M.visible_message(span_warning("[M]'s [A] glows brightly as it reflects the power back into the hammer!"))
+			visible_message(span_boldwarning("<span class='big bold'>The power swelling inside Mjolnir becomes too much and its form fractures!</span><br>"))
+			explosion(src, 0, 1, 5, 5, TRUE, FALSE, 2)
+			playsound(src, 'sound/magic/lightningbolt.ogg', 100, 1, extrarange = 30)
+			tesla_zap(src, tesla_range, tesla_power, tesla_flags)
```
Comment (ToasterBiome) [inline code/game/objects/items/singularityhammer.dm]: yeah these too whatever

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

**Review by ToasterBiome — APPROVED** [review]

**Comment by alexkar598** [comment]
So it takes 1 guy with a bible you can't see and you lose 2 spell points?

**Comment by ToasterBiome** [comment]
> So it takes 1 guy with a bible you can't see and you lose 2 spell points?

don't throw your hammer at the guy who is anti magic dum dum

**Comment by Marmio64** [comment]
this sounds pretty lame

instead, why not just implement a cooldown window for ranged stuns and if it impacts an antimagic person it procs the cooldown while also reflecting the hammer.

or maybe instead if it hits an antimagic person it reflects the hammer back into the wizard and smacks them instead.

edit: also, you can counter lich + recall mjolnir strat by just deepfrying the hammer after the lich dies

**Comment by ToasterBiome** [comment]
These are all cool ideas, up to HC to tell @Chubbygummibear what to do with it

**Comment by alexkar598** [comment]
> > So it takes 1 guy with a bible you can't see and you lose 2 spell points?
> 
> don't throw your hammer at the guy who is anti magic dum dum

Ok, from these 5 people, tell me who jumped the library counter and printed a bible?

**Comment by ToasterBiome** [comment]
> > > So it takes 1 guy with a bible you can't see and you lose 2 spell points?
> > 
> > 
> > don't throw your hammer at the guy who is anti magic dum dum
> 
> Ok, from these 5 people, tell me who jumped the library counter and printed a bible?

Why the fuck is bible antimagic for non-chaplains

**Comment by ToasterBiome** [comment]
> > > So it takes 1 guy with a bible you can't see and you lose 2 spell points?
> > 
> > 
> > don't throw your hammer at the guy who is anti magic dum dum
> 
> Ok, from these 5 people, tell me who jumped the library counter and printed a bible?

https://github.com/yogstation13/Yogstation/pull/13264

**Comment by Chubbygummibear** [comment]
> this sounds pretty lame
> 
> instead, why not just implement a cooldown window for ranged stuns and if it impacts an antimagic person it procs the cooldown while also reflecting the hammer.
> 
> or maybe instead if it hits an antimagic person it reflects the hammer back into the wizard and smacks them instead.
> 
> edit: also, you can counter lich + recall mjolnir strat by just deepfrying the hammer after the lich dies

After you said this I tried to do the hammer reflect using boomerang code since i wanted something that already had a "returning" functionality, but it failed and the hammer did not rethrow back, and then I thought that was weird so i spawned in a boomerang to see it in action only to find it doesn't work itself, so that's cool. I'll try to figure it out if HC doesn't like my proposal, but as it stands I still like hammersplosion

**Comment by patpol4** [comment]
Shouldn't make you lose the hammer, that part is stupid

**Comment by JamieD1** [comment]
How about if it hits an anti magic person the hammer lands on the ground unpickup-able and has a cooldown of around 30/60 seconds?

So it sucks but its not a waste of points

**Comment by Chubbygummibear** [comment]
> How about if it hits an anti magic person the hammer lands on the ground unpickup-able and has a cooldown of around 30/60 seconds?
> 
> So it sucks but its not a waste of points

acknowledged and working on it
image

**Comment by Chubbygummibear** [comment]
https://user-images.githubusercontent.com/46236974/153739839-b76414d2-3bf9-40f7-a508-c8cfcad5e19c.mp4

Thots? i just realized there's league noises in the background lmao

**Comment by Marmio64** [comment]
a 30 second pullout time is stupid, you might as well lose the hammer because if your hammer gets stuck while you're in the halls that shit is never going to be pulled out in 30 seconds without some assistant walking up to you and pushing you. I def do not think he meant by cooldown time of 30/60 seconds that you would have to stand still the entire duration. 10 or 15 seconds seems much more reasonable.

**Comment by Moja10** [comment]
Original PR was ultra cringe, but the pullout of 5 - 10 seconds embed in ground is good. However, I never really found the mjolnir recall lich stuff that problematic. Also, we should be back summon guns, thank you.

File: code/game/objects/structures/mjollnir.dm [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/obj/structure/mjollnir
+	name = "Mjolnir"
+	desc = "This once powerful hammer feels cold to the touch, but a thrum of magic inside tells you its only a matter of time until it awakens again."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "mjollnir-inert"
+	resistance_flags = INDESTRUCTIBLE
+	anchored = TRUE
+	density = FALSE
+	var/pull_time = 150
```
Comment (ToasterBiome) [inline code/game/objects/structures/mjollnir.dm]:
```suggestion
	var/pull_time = 15 SECONDS
```

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

**Comment by patpol4** [comment]
Why not literally just have it do no damage

**Comment by bruhlookatthisdood** [comment]
so what if you dont throw it and just whack them?

**Comment by Chubbygummibear** [comment]
> Why not literally just have it do no damage

I wanted it to be some kind of fantastical interaction because what I think is the pinnacle of magic vs anti-magic interaction is disintegrate's anti-magic interaction where if you cast it on anti-magic, your whole arm blows off. That's an appropriate drawback for what would otherwise be an instant kill effect. Mjolnir is not an instant kill, but it is a guaranteed 6 second stun on hit. That paired with the throw, and 30 brute throw force makes it incredibly safe to fling around and in many cases landing one hit is all it takes to ensure someone's death with a few follow up smacks. While I will say the explosion was probably too punishing, i do want some kind of punishing effect for throwing the hammer (taking the safe option) into anti-magic.

Also I was inspired by thor losing his hammer by throwing it lmao
mjolnir destroyed



> so what if you dont throw it and just whack them?

untouched, smacking up close is fine and won't cause you to lose the hammer in any way

File: code/game/objects/items/singularityhammer.dm [hunk 6]
```diff
@@ -85,6 +85,9 @@
 	throwforce = 30
 	throw_range = 7
 	w_class = WEIGHT_CLASS_HUGE
+	var/zap_power = 20000
```
Comment (ToasterBiome) [inline code/game/objects/items/singularityhammer.dm]: these don't look like they're used anymore

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

**Comment by Hopekz** [comment]
> don't throw your hammer at the guy who is anti magic dum dum

Anyone can have anti magic. There's no real way to tell if someone has something in their pockets. Your logic is flawed because it assumes that they'll always be plainly visible.

 Even if you're saying to just not throw it at someone that looks like a chaplain there is nothing preventing the chaplain from looking like a regular crew member.

**Comment by Hopekz** [comment]
Also @Chubbygummibear stop making it behave like in the movies and using movies to justify it since this is supposed to be ss13 and a slight reference vs being a 1:1 copy

**Comment by Marmio64** [comment]
besides, that lady was like, a god or something, and she crushed the hammer in her hands rather than it being demolished from hitting her. The chaplains just some bloodthirsty preacher with a holy artifact that can repel dark magics.

File: code/game/objects/items/singularityhammer.dm (hunk 6 above)
Comment (Chubbygummibear) [reply code/game/objects/items/singularityhammer.dm]: you right, they gone now

**Comment by Chubbygummibear** [comment]
the exploding hammer from Thor was the initial inspiration for what to do with the hammer, but don't worry, we've moved away from the hammer exploding entirely. Even though i still like it and think it's cool 😔

**Comment by JamieD1** [comment]
> the exploding hammer from Thor was the initial inspiration for what to do with the hammer, but don't worry, we've moved away from the hammer exploding entirely. Even though i still like it and think it's cool 😔

Could make a wizard suplexing the hammer break it. Obviously this would mainly be wizard vs wizard

**Comment by Chubbygummibear** [comment]
> > the exploding hammer from Thor was the initial inspiration for what to do with the hammer, but don't worry, we've moved away from the hammer exploding entirely. Even though i still like it and think it's cool 😔
> 
> Could make a wizard suplexing the hammer break it. Obviously this would mainly be wizard vs wizard

Would that be possible considering the hammer doesn't move in the same fashion the rod does? i would maybe do if the throw connects with a wizard, but that seems along the same level of unfair as the hammersplosion because it wouldn't even require antimagic, just to exist as a wizzie and get hit. I think I might just commit to the hammer sinking to the ground and becoming a statue since it accomplishes the nerfs of how safe throwing it is while having recall ready. Now you have to take a moment to check your targets, and if you still throw it about without regard, you'll lose it temporarily and break the recall. If you have any requirements you'd like to see for what i have now just let me know 👌

**Comment by KittyNoodle** [comment]
I feel like you are missing the obvious solution here is to just make the hammer be catchable by people with anti-magic

**Review by nesisknesis — APPROVED** [review]

**Comment by JamieD1** [comment]
> I feel like you are missing the obvious solution here is to just make the hammer be catchable by people with anti-magic

UnIronically a good idea

**Review by ToasterBiome — APPROVED** [review]

**Comment by JamieD1** [comment]
> > I feel like you are missing the obvious solution here is to just make the hammer be catchable by people with anti-magic
> 
> UnIronically a good idea

Thoughts on this? ^

**Comment by Chubbygummibear** [comment]
> > > I feel like you are missing the obvious solution here is to just make the hammer be catchable by people with anti-magic
> > 
> > 
> > UnIronically a good idea
> 
> Thoughts on this? ^

sorry for not replying sooner, i really wasn't sure if that was a prompt for a response or just a comment. I have a few issues with catching it as the anti magic effect. Namely, that i don't find just being able to catch it to be interesting or wacky. my basis for an anti magic effect that i like is the disintegrate backfire that blows the caster's arm off if you touch someone with anti magic. that's cool, impressive, and wacky. i feel like magic is best captured as an element when the results are wild and weird. additionally to this, changing the effect to catching doesn't disrupt the throw recall combo that i was initially setting out to try and nerf. i was looking for a way to wipe the recall mark from mjolnir at the same time by way of manipulating the hammer as part of the backfire effect, and i'm not sure if i could wipe that mark without deleting the hammer and putting it into the catcher's hands. also speaking of hands, i'm not too sure how catching works, but it's my understanding that you need to have throw intent on with an open hand to catch, which means your other hand would have to be holding your anti magic item unless you somehow had the wizard hardsuit, or a tinfoil hat. this sounds like you would have to have anti magic in one hand, and the other empty with throw intent on to actually catch the hammer, effectively meaning both hands are taken and you cannot do anything otherwise like shooting back, or holding a weapon. with the recent pr pointing out that holy anti magic from the bible doesn't work in most regular magic cases, that means the number of anti magic sources without catches is pretty slim. you either have the chaplain ready to combat the wizard with his null rod, or you wait 10 minutes for botany to produce holy melons. 
also i was kinda hoping i wouldn't have to rework it a 3rd time 😔

**Review by JamieD1 — APPROVED** [review]

**Comment by Chubbygummibear** [comment]
fuck sorry jamie i should have reworded the changelog before you merged it so people don't get the wrong idea about what's happening. is that still fixable?

**Pull request merged**